components/wistia/sources/new-project/new-project.mjs

```diff
@@ -0,0 +1,54 @@
+import wistia from "../../wistia.app.mjs";
+
+export default {
+  name: "New Project",
+  version: "0.0.1",
+  key: "wistia-new-project",
+  description: "Emit new event for each created project.",
+  type: "source",
+  dedupe: "unique",
+  props: {
+    wistia,
+  },
+  methods: {
+    emitEvent(data) {
+      this.$emit(data, {
+        id: data.id,
+        summary: `New project uploaded with id ${data.id}`,
+        ts: Date.parse(data.created),
+      });
+    },
+  },
+  hooks: {
+    async deploy() {
+      const projects = await this.wistia.getProjects({
+        params: {
+          per_page: 10,
+        },
+      });
+
+      projects.forEach(this.emitEvent);
+    },
+  },
+  async run() {
+    let page = 0;
+
+    while (page >= 0) {
```

Comment:
Won't this always start from page 0 in every run?
Can't we avoid making unnecessary requests?
Also, remember that `unique` deduping [only caches the last 100 ids](https://pipedream.com/docs/components/api/#dedupe-strategies).

Reply:
Yes, because user can delete medias.
For example, I have 1500 (page 15) medias uploaded, and I delete 500 and after I upload more 50,. Source will need start from page 0 to get the new 50 uploaded medias.